### PR TITLE
#171: Updated poetry dependencies

### LINKED ISSUE #171: Update poetry dependencies
Fixes https://github.com/exasol/data-science-utils-python/issues/71, too

General strategy: Try to modify `pyproject.toml` to use latest version

* Dependency to `exasol-data-science-utils`
  * should be removed, see issue [DSUP #71](https://github.com/exasol/data-science-utils-python/issues/71)
  * Copied code into AAF [unit_tests/data_science_utils](/github.com/exasol/advanced-analytics-framework/tree/refactoring/%23171-Updated_poetry_dependencies/tests/unit_tests/data_science_utils)
  * Copied tests into AAF [tests/integration_tests/data_science_utils](https://github.com/exasol/advanced-analytics-framework/tree/refactoring/%23171-Updated_poetry_dependencies/tests/integration_tests/data_science_utils) `test_pyexasol_sql_executor.py`
* UDP exasol-udf-mock-python: replace git dependency by dependency to latest pypi version [0.1.0](https://pypi.org/project/exasol-udf-mock-python/0.1.0/)

Special cases
* Build scripts
  * Currently dynamic matrix
  * Some tests have rather long duration
* :white_check_mark: Update pyexasol
* typeguard update will cause problems
  * see existing branch [tensorflow_example](https://github.com/exasol/advanced-analytics-framework/tree/tensorflow_example)
* :warning: Update bucketfs-python to [0.11.0](https://pypi.org/project/exasol-bucketfs/0.11.0/) on pypi (currently 0.8.0 due to incompatibilities)
* Update itde (can get problematic because of the changed ports)
  * Proposal: use new port conventions in AAF
* Update to python 3.10
  * `conda.yml`
  * ... (more)

### PR DESCRIPTION
**Pull request by ckunki**
Copied subdirectories from data-science-utils-python into subdirectory exasol_data_science_utils_python.

closes #171 
closes https://github.com/exasol/data-science-utils-python/issues/71

### CODE REVIEW

File: tests/unit_tests/data_science_utils/utils/test_hash_generation_for_object.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/utils/test_hash_generation_for_object.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/udf_utils/test_udf_context_wrapper.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/udf_utils/test_udf_context_wrapper.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/udf_utils/test_iterate_trough_dataset.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/udf_utils/test_iterate_trough_dataset.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/udf_utils/test_ctx_iterator.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/udf_utils/test_ctx_iterator.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/test_mock_sql_executor.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/test_mock_sql_executor.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/test_mock_result_set.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/test_mock_result_set.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_view_name.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_view_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_view.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_view.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_udf_name_builder.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_udf_name_builder.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_udf_name.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_udf_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_table_name_builder.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_table_name_builder.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_table_name.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_table_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_table_builder.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_table_builder.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_table.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_table.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_schema_name.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_schema_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_identifier.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_identifier.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_connection_object_name_builder.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_connection_object_name_builder.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_connection_object_name.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_connection_object_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_column_type.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_column_type.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_column_name_builder.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_column_name_builder.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_column_name.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_column_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_column_builder.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_column_builder.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: tests/unit_tests/data_science_utils/schema/test_column.py
Comment (ckunki) [inline tests/unit_tests/data_science_utils/schema/test_column.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/utils/hash_generation_for_object.py
Comment (ckunki) [inline exasol_data_science_utils_python/utils/hash_generation_for_object.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/utils/data_classes_runtime_type_check.py
Comment (ckunki) [inline exasol_data_science_utils_python/utils/data_classes_runtime_type_check.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/udf_utils/udf_context_wrapper.py
Comment (ckunki) [inline exasol_data_science_utils_python/udf_utils/udf_context_wrapper.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/udf_utils/testing/mock_sql_executor.py
Comment (ckunki) [inline exasol_data_science_utils_python/udf_utils/testing/mock_sql_executor.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/udf_utils/testing/mock_result_set.py
Comment (ckunki) [inline exasol_data_science_utils_python/udf_utils/testing/mock_result_set.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/udf_utils/sql_executor.py
Comment (ckunki) [inline exasol_data_science_utils_python/udf_utils/sql_executor.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/udf_utils/pyexasol_sql_executor.py
Comment (ckunki) [inline exasol_data_science_utils_python/udf_utils/pyexasol_sql_executor.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/udf_utils/iterator_utils.py
Comment (ckunki) [inline exasol_data_science_utils_python/udf_utils/iterator_utils.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/view_name_impl.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/view_name_impl.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/view_name_builder.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/view_name_builder.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/view_name.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/view_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/view.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/view.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/udf_name_impl.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/udf_name_impl.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/udf_name_builder.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/udf_name_builder.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/udf_name.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/udf_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/table_name_impl.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/table_name_impl.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/table_name_builder.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/table_name_builder.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/table_name.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/table_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/table_like_name_impl.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/table_like_name_impl.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/table_like_name.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/table_like_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/table_like.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/table_like.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/table_builder.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/table_builder.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/table.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/table.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/schema_name.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/schema_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/experiment_name.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/experiment_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/exasol_identifier_impl.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/exasol_identifier_impl.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/exasol_identifier.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/exasol_identifier.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/dbobject_name_with_schema_impl.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/dbobject_name_with_schema_impl.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/dbobject_name_with_schema.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/dbobject_name_with_schema.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/dbobject_name_impl.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/dbobject_name_impl.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/dbobject_name.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/dbobject_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/dbobject.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/dbobject.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/connection_object_name_impl.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/connection_object_name_impl.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/connection_object_name_builder.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/connection_object_name_builder.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/connection_object_name.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/connection_object_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/column_type.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/column_type.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/column_name_builder.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/column_name_builder.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/column_name.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/column_name.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/column_builder.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/column_builder.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: exasol_data_science_utils_python/schema/column.py
Comment (ckunki) [inline exasol_data_science_utils_python/schema/column.py:1]: File copied from [data-science-utils-python](https://github.com/exasol/data-science-utils-python/)

File: pyproject.toml [hunk 1]
```diff
@@ -13,32 +18,33 @@ readme = 'README.md'
 repository = "https://github.com/exasol/advanced-analytics-framework"
 homepage = "https://github.com/exasol/advanced-analytics-framework"
 
-
 keywords = ['exasol']
 
+# typeguard                                  2.13.3    4.3.0
```
Comment (tkilias) [inline pyproject.toml:23]: Is this still needed

File: pyproject.toml [hunk 2]
```diff
@@ -13,32 +18,33 @@ readme = 'README.md'
...
-pytest = "^7.1.2"
-poethepoet = "^0.13.0"
+pytest = ">=7.2.2"
 pytest-assume = "^2.4.3"
-exasol-udf-mock-python = { git = "https://github.com/exasol/udf-mock-python.git", branch = "main" }
-pytest-cov = "^3.0.0"
-exasol-integration-test-docker-environment = "1.7.1"
+exasol-udf-mock-python = ">=0.2.0,<1.0.0"
+pytest-cov = "^5.0.0"
+# exasol-integration-test-docker-environment = "1.7.1"
```
Comment (tkilias) [inline pyproject.toml:46]: Is this still needed

File: tests/integration_tests/data_science_utils/test_pyexasol_sql_executor.py
Comment (tkilias) [inline tests/integration_tests/data_science_utils/test_pyexasol_sql_executor.py:1]: File was copied

File: pyproject.toml (hunk 2 above)
Comment (ckunki) [reply pyproject.toml]: See next push - removed

File: pyproject.toml (hunk 1 above)
Comment (ckunki) [reply pyproject.toml]: removed in next push.

**Review by tkilias — APPROVED** [review]

**Pull request merged**